job_templates/capsule_upgrade_-_ansible_default.erb

```diff
@@ -0,0 +1,70 @@
+<%#
+name: Capsule Upgrade Playbook
+snippet: false
+template_inputs:
+- name: target_version
+  required: true
+  input_type: user
+  advanced: false
+  value_type: plain
+  hidden_value: false
+- name: whitelist_options
+  required: false
+  input_type: user
+  advanced: false
+  value_type: plain
+  hidden_value: false
+model: JobTemplate
+job_category: Ansible Playbook
+description_format: "%{template_name}"
+provider_type: Ansible
+kind: job_template
+%>
+
+---
+- hosts: all
+  tasks:
+    - name: Gather the rpm package facts
+      package_facts:
+        manager: auto
+
+    - name: Fail if the target server is a Satellite server
+      fail:
+        msg: "This playbook cannot be executed on a Satellite server. Use only on a Capsule server."
+      when: "'satellite' in ansible_facts.packages"
+
+    - name: Install foreman-maintain if not present
+      package:
+        name: rubygem-foreman_maintain
+        state: present
+      when: "'rubygem-foreman_maintain' not in ansible_facts.packages"
+
+    - block:
+      - name: Upgrade Capsule server using foreman-maintain
+        shell: foreman-maintain upgrade run --assumeyes --target-version=<%= input('target_version') %> --whitelist="<%= input('whitelist_options') %>"
```

Comment:
Is it possible to omit the --whitelist option from the command in case no whitelist_options were provided? Otherwise the foreman-maintain would complain with "ERROR: option '--whitelist': value not specified".

Reply:
Good catch, can you please create a foreman-issue to track?

Reply:
Yes, sure, here we go: https://projects.theforeman.org/issues/29616